runtime/near-vm-logic/src/tests/bls12_381_verify.rs

```diff
@@ -0,0 +1,476 @@
+// the tests from this repo are used https://github.com/ethereum/bls12-381-tests
+use crate::map;
+use crate::tests::fixtures::get_context;
+use crate::tests::helpers::assert_costs;
+use crate::tests::vm_logic_builder::VMLogicBuilder;
+use hex::FromHex;
+use near_primitives_core::config::ExtCosts;
+use near_vm_errors::HostError::Bls1238VerifyError;
+use near_vm_errors::{Bls12381Error, HostError, VMLogicError};
+use std::collections::HashMap;
+
+#[track_caller]
+fn check_bls12_381_verify(
+    signature_len: u64,
+    signature: &[u8],
+    message_len: u64,
+    message: &[u8],
+    public_key_len: u64,
+    public_key: &[u8],
+    want: Result<u64, HostError>,
+    want_costs: HashMap<ExtCosts, u64>,
+) {
+    let mut logic_builder = VMLogicBuilder::default();
+    let mut logic = logic_builder.build(get_context(vec![], false));
+
+    let signature_ptr = if signature_len == u64::MAX {
+        logic.wrapped_internal_write_register(1, &signature).unwrap();
+        1
+    } else {
+        signature.as_ptr() as u64
+    };
+
+    let message_ptr = if message_len == u64::MAX {
+        logic.wrapped_internal_write_register(2, &message).unwrap();
+        2
+    } else {
+        message.as_ptr() as u64
+    };
+
+    let public_key_ptr = if public_key_len == u64::MAX {
+        logic.wrapped_internal_write_register(3, &public_key).unwrap();
+        3
+    } else {
+        public_key.as_ptr() as u64
+    };
+
+    let result = logic.bls12_381_aggregate_verify(
+        signature_ptr,
+        signature_len,
+        message_ptr,
+        message_len,
+        public_key_ptr,
+        public_key_len,
+    );
+
+    let want = want.map_err(VMLogicError::HostError);
+    assert_eq!(want, result);
+    assert_costs(want_costs);
+}
+
+#[test]
+fn test_bls12_381_verify_valid() {
+    let hex_convert_error_msg = "Error during converting hex string to bytes";
+    let signature: Vec<u8> = <Vec<u8>>::from_hex("912c3615f69575407db9392eb21fee18fff797eeb2fbe1816366ca2a08ae574d8824dbfafb4c9eaa1cf61b63c6f9b69911f269b664c42947dd1b53ef1081926c1e82bb2a465f927124b08391a5249036146d6f3f1e17ff5f162f779746d830d1").expect(hex_convert_error_msg);
```

Comment:
would be nice to have a reference, or a way to reproduce these signatures

Reply:
fixed: https://github.com/near/nearcore/pull/8184/commits/8a083a29eb5ed8829ea42035c39c3c70c6fa1d83